hash.go

```diff
@@ -85,21 +90,24 @@ func addGarbleToHash(inputHash []byte) []byte {
 	// Join the two content IDs together into a single base64-encoded sha256
 	// sum. This includes the original tool's content ID, and garble's own
 	// content ID.
-	h := sha256.New()
-	h.Write(inputHash)
+	hasher.Reset()
+	hasher.Write(inputHash)
 	if len(cache.BinaryContentID) == 0 {
 		panic("missing binary content ID")
 	}
-	h.Write(cache.BinaryContentID)
+	hasher.Write(cache.BinaryContentID)
 
 	// We also need to add the selected options to the full version string,
 	// because all of them result in different output. We use spaces to
 	// separate the env vars and flags, to reduce the chances of collisions.
 	if cache.GOGARBLE != "" {
-		fmt.Fprintf(h, " GOGARBLE=%s", cache.GOGARBLE)
+		fmt.Fprintf(hasher, " GOGARBLE=%s", cache.GOGARBLE)
 	}
-	appendFlags(h, true)
-	return h.Sum(nil)[:buildIDComponentLength]
+	appendFlags(hasher, true)
+	// addGarbleToHash returns the sum buffer, so we need a new copy.
+	// Otherwise the next use of the global sumBuffer would conflict.
+	sumBuffer := make([]byte, 0, sha256.Size)
```

Comment:
Maybe rename this variable so it's obvious it's a new slice and unrelated to the global `sumBuffer` array?

Reply:
I thought about this for a bit, actually - I'm not sure. I ended up with sumBuffer because then we shadow the global one, ensuring we don't use the global one in the following lines.

Reply:
Hmm, that's an interesting reason for shadowing... haven't thought of it like that before. I don't feel very strongly about this, and as shadowing `sumBuffer` prevents use of the global one, as you said I think it's fine to leave as is.

Reply:
I admit this could go either way :)